lib/maths/CBoostedTreeImpl.cc

```diff
@@ -318,6 +318,10 @@ CBoostedTreeImpl::trainForest(core::CDataFrame& frame,
             forest.push_back(std::move(tree));
         }
         LOG_TRACE(<< "bias = " << (1.0 - oneMinusBias));
+
+        if (m_Loss->isCurvatureConstant() == false) {
```

Comment:
Maybe I missed something, but `m_Loss->isCurvatureConstant() == false` will always be `false` since the only implementation has a constant return value of `true`. 

So within `trainForest`, we only ever calculate the candidateSplits once?

Reply:
It is always a constant for MSE. In general, it doesn't have to be. However, in this case there is no need to recompute splits we can get away with using the same candidates. The candidates are chosen to be fairly fine grained so selecting splits when building the tree from this set means the optimal (binary tree) should close to one that is actually available.